projects/dotcms/src/lib/utils/request.ts

```diff
@@ -28,7 +28,7 @@ async function getUrl(
 }
 
 function shouldAppendBody(params: DotAppHttpRequestParams): boolean {
-    return params.method === 'POST' && !!params.body;
+    return (params.method === 'POST' || params.method === 'PUT') && !!params.body;
```

Comment:
shouldn't include patch too?

Reply:
We could pass a body with GET method also. I would have added an overload for each request method and accept the right parameters. Feel free to add every method your application uses or design the http class differently.

I ended replacing node fetch by axios.